### PR TITLE
VY-AUTH-39 -Integrate Google Oauth

### PR DESCRIPTION
**Pull request by lf-achyutpkl**
@anishmanandhar  please review my code

### CODE REVIEW

File: src/main/java/com/lftechnology/vyaguta/auth/exception/TokenExpiredException.java [hunk 1]
```diff
@@ -3,10 +3,18 @@
 /**
  * 
  * @author Kailash Bijayananda <kailashraj@lftechnology.com>
- *
+ * @author Achyut Pokhrel <achyutpokhrel@lftechnology.com>
  */
-public class TokenExpiredException extends Exception {
+public class TokenExpiredException extends RuntimeException {
```
Comment (anishmanandhar) [inline src/main/java/com/lftechnology/vyaguta/auth/exception/TokenExpiredException.java]: Its good to extends Our custom business exception as run time :+1:

File: src/main/java/com/lftechnology/vyaguta/auth/rs/AuthRs.java [hunk 2]
```diff
@@ -41,66 +40,37 @@
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     public Response token(String jsonString) {
-        try {
-            LoginDTO dto = userService.buildLoginDTO(jsonString);
+        LoginDTO dto = userService.buildLoginDTO(jsonString);
```
Comment (anishmanandhar) [inline src/main/java/com/lftechnology/vyaguta/auth/rs/AuthRs.java]: Seems like UserService buildLoginDto is not only building LoginDTO

**Comment by bhuwang** [comment]
Can one of the admins verify this patch?

**Comment by bhuwang** [comment]
Can one of the admins verify this patch?

**Comment by bhuwang** [comment]
Can one of the admins verify this patch?

File: frontend/src/js/util/ApiSession.js [hunk 3]
```diff
@@ -13,11 +14,26 @@
...
+        },
+
+        tokenLogin : function (token, callback){
+            request
+                .post(url + urlConstants.token)
+                .send({grant_type: 'google', token: token})
+                .set('Accept', 'application/json')
+                .end(function(err, res){
+                    if(!err){
+                        console.log("response" + res.body);
```
Comment (lf-kailashraj) [inline frontend/src/js/util/ApiSession.js]: ?

File: src/main/java/com/lftechnology/vyaguta/auth/service/UserServiceImpl.java [hunk 4]
```diff
@@ -14,34 +15,48 @@
...
 public class UserServiceImpl implements UserService {
 
+    private static final String ID = "id";
+    private static final String HD = "hd";
+    private static final String NAME = "name";
+    private static final String EMAIL = "email";
+    private static final String ADMIN = "Admin";
+    private static final String PICTURE = "picture";
+    private static final String GRANT_TYPE = "grant_type";
+    private static final String LFTECHNOLOGY_COM = "lftechnology.com";
```
Comment (lf-kailashraj) [inline src/main/java/com/lftechnology/vyaguta/auth/service/UserServiceImpl.java]: thukkka

File: src/main/java/com/lftechnology/vyaguta/auth/service/UserServiceImpl.java [hunk 5]
```diff
@@ -50,10 +65,15 @@
...
     private GoogleOAuthService googleOAuthService;
 
+    @Inject
+    RoleDao roleDao;
+
     @Override
     public User save(User entity) {
-        String saltedPassword = salt + entity.getPassword();
-        entity.setPassword(DigestUtils.shaHex(saltedPassword));
+        if (entity.getAuthProvider() != User.AUTH_PROVIDER_GOOGLE) {
```
Comment (lf-kailashraj) [inline src/main/java/com/lftechnology/vyaguta/auth/service/UserServiceImpl.java]: ???

File: src/main/java/com/lftechnology/vyaguta/auth/service/UserServiceImpl.java [hunk 6]
```diff
@@ -14,34 +15,48 @@
...
  * 
  * @author Kailash Bijayananda <kailashraj@lftechnology.com>
+ * @author Achyut Pokhrel <achyutpokhrel@lftechnology.com>
  *
  */
 @Stateless
 public class UserServiceImpl implements UserService {
 
+    private static final String ID = "id";
+    private static final String HD = "hd";
```
Comment (lf-kailashraj) [inline src/main/java/com/lftechnology/vyaguta/auth/service/UserServiceImpl.java]: ?

**Pull request merged**